src/kvstore/NebulaStore.cpp

```diff
@@ -533,6 +533,74 @@ ResultCode NebulaStore::flush(GraphSpaceID spaceId) {
     return ResultCode::SUCCEEDED;
 }
 
+ResultCode NebulaStore::createCheckpoint(GraphSpaceID spaceId, const std::string& name) {
+    auto spaceRet = space(spaceId);
+    if (!ok(spaceRet)) {
+        return error(spaceRet);
+    }
+
+    auto space = nebula::value(spaceRet);
+    for (auto& engine : space->engines_) {
+        auto code = engine->createCheckpoint(name);
+        if (code != ResultCode::SUCCEEDED) {
+            return code;
+        }
+        // create wal hard link for all parts
+        auto parts = engine->allParts();
+        for (auto& part : parts) {
+            auto ret = this->part(spaceId, part);
+            if (!ok(ret)) {
+                LOG(ERROR) << "Part not found. space : " << spaceId << " Part : " << part;
+                return error(ret);
+            }
+            auto walPath = folly::stringPrintf("%s/checkpoints/%s/wal/%d",
+                                                      engine->getDataRoot(), name.c_str(), part);
+            auto p = nebula::value(ret);
+            if (!p->linkCurrentWAL(walPath.data())) {
+                return ResultCode::ERR_CHECKPOINT_ERROR;
+            }
+        }
+    }
+    return ResultCode::SUCCEEDED;
+}
+
+ResultCode NebulaStore::dropCheckpoint(GraphSpaceID spaceId, const std::string& name) {
+    auto spaceRet = space(spaceId);
+    if (!ok(spaceRet)) {
+        return error(spaceRet);
+    }
+    auto space = nebula::value(spaceRet);
+    for (auto& engine : space->engines_) {
+        auto code = engine->dropCheckpoint(name);
+        if (code != ResultCode::SUCCEEDED) {
+            return code;
+        }
+    }
+    return ResultCode::SUCCEEDED;
+}
+
+ResultCode NebulaStore::setPartBlocking(GraphSpaceID spaceId, PartitionID partId, bool sign) {
+    folly::RWSpinLock::ReadHolder rh(&lock_);
```

Comment:
You could call part(spaceId, partId) directly.

Reply:
excellent!